File: c#/src/v1/models/ApiResponse.cs
using System.Text.Json.Serialization;

namespace Comms.Models;

public class ApiResponse
{
    [JsonPropertyName("Status")]
    public ApiResponseCode Status { get; set; }

    [JsonPropertyName("Message")]
    public string? Message { get; set; }

    [JsonPropertyName("Cost")]
    public int? Cost { get; set; }

    [JsonPropertyName("Currency")]
    public string? Currency { get; set; }

    [JsonPropertyName("MsgFollowUpUniqueCode")]
    public string? MessageFollowUpCode { get; set; }

    [JsonPropertyName("Balance")]
    public double? Balance { get; set; }
}

File: kotlin/src/main/kotlin/v1/models/ApiResponse.kt
package v1.models

import com.fasterxml.jackson.annotation.JsonProperty

class ApiResponse {
    @JsonProperty(required = true, value = "Status")
    var status: ApiResponseCode? = null

    @JsonProperty(value = "Message")
    var message: String? = null

    @JsonProperty(value = "Cost")
    var cost: Int? = null

    @JsonProperty(value = "MsgFollowUpUniqueCode")
    var messageFollowUpCode: String? = null

    @JsonProperty(value = "Balance")
    var balance: Double? = null

}
